pulsar-broker/src/main/java/org/apache/pulsar/broker/lookup/TopicLookupBase.java

```diff
@@ -342,6 +343,22 @@ public static CompletableFuture<ByteBuf> lookupTopicAsync(PulsarService pulsarSe
         return lookupfuture;
     }
 
+    /**
+     * Check if a internal client will get a null lookup result.
+     */
+    private static void printWarnLogIfLookupResUnexpected(TopicName topic, LookupData lookupData,
+                                                          PulsarService pulsar) {
+        if (!pulsar.getBrokerService().isSystemTopic(topic)) {
+            return;
+        }
+        boolean tlsEnabled = pulsar.getConfig().isBrokerClientTlsEnabled();
+        if (!tlsEnabled && StringUtils.isBlank(lookupData.getBrokerUrl())) {
+            log.warn("[{}] Unexpected lookup result {}", topic.toString(), lookupData);
```

Comment:
It would be useful to improve the log message to be explicit about why it's unexpected by adding "brokerUrl is required when TLS is disabled."

Reply:
Improved